code/modules/projectiles/updated_projectiles/gun_helpers.dm

```diff
@@ -342,9 +342,26 @@ should be alright.
 		to_chat(user, "<span class='warning'>The attachment on [src]'s [attachment.slot] cannot be removed!</span>")
 		return
 
-	user.visible_message("<span class='notice'>[user] begins attaching [attachment] to [src].</span>",
-	"<span class='notice'>You begin attaching [attachment] to [src].</span>", null, 4)
-	if(do_after(user,60, TRUE, 5, BUSY_ICON_FRIENDLY))
+
+	var/attach_delay = 30
+	var/firearm_skill = user.mind.cm_skills.firearms
+	if(attachment.name == "bayonet" && (!src.muzzle || src.muzzle.name == "bayonet")) //Attach  bayonet fast if no attachment to remove on the barrel, or you're replacing a bayo with another bayo
+		attach_delay = 10
```

Comment:
If you want different delays based on attachment it needs to be an attachment var